plugins/plugin_utils/lookup.py

```diff
@@ -12,109 +12,90 @@
-                "resource_pools": "resource_pools",
-            }
-        }
+        "parent_resource_pools": "parent_resource_pools",
+        "resource_pools": "resource_pools",
     },
     "datacenter": {
-        "list": {
-            "query": {
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "names": "names",
-            }
-        }
-    },
-    "folder": {
-        "list": {
-            "query": {
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "names": "names",
-                "parent_folders": "parent_folders",
-                "type": "type",
-            }
-        }
+        "parent_folders": "folders",
     },
+    "folder": {},
     "cluster": {
-        "list": {
-            "query": {
-                "clusters": "clusters",
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "names": "names",
-            }
-        }
+        "parent_folders": "folders",
     },
     "host": {
-        "list": {
-            "query": {
-                "clusters": "clusters",
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "hosts": "hosts",
-                "names": "names",
-            }
-        }
+        "parent_folders": "folders",
     },
     "datastore": {
-        "list": {
-            "query": {
-                "datacenters": "datacenters",
-                "datastores": "datastores",
-                "folders": "folders",
-                "names": "names",
-                "types": "types",
-            }
-        }
+        "parent_folders": "folders",
     },
     "vm": {
-        "list": {
-            "query": {
-                "clusters": "clusters",
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "hosts": "hosts",
-                "names": "names",
-                "resource_pools": "resource_pools",
-                "vms": "vms",
-            }
-        }
-    },
-    "network": {
-        "list": {
-            "query": {
-                "datacenters": "datacenters",
-                "folders": "folders",
-                "names": "names",
-                "networks": "networks",
-                "types": "types",
-            }
-        }
+        "parent_folders": "folders",
     },
+    "network": {"parent_folders": "folders"},
 }
 
 
+class VcenterApi:
+    def __init__(self, hostname, session):
+        self.hostname = hostname
+        self.session = session
+
+    async def fetch(self, url):
+        async with self.session.get(url) as response:
+            result = await response.json()
+            return result
+
+    def build_url(self, object_type, filters):
+        corrected_filters_for_query = self.correct_filter_names(filters, object_type)
+        if object_type == "resource_pool":
```

Comment:
Maybe worth to make it generic?
Like `if '_' in object_type`?

Reply:
good point, i think theres no reason to even have the `if` statement too